materialized_sql_view/model/abstract_materialized_sql_view.py

```diff
@@ -0,0 +1,336 @@
+        """
+
+    def after_drop_materialized_view(self, cr, uid, context=None):
+        """Method called after drop materialized view and view,
+           Nothing done in abstract method, it's  hook to used in subclass
+        """
+
+    def before_create_materialized_view(self, cr, uid, context=None):
+        """Method called before create materialized view and view,
+           Nothing done in abstract method, it's  hook to used in subclass
+        """
+
+    def after_create_materialized_view(self, cr, uid, context=None):
+        """Method called after create materialized view and view,
+           Nothing done in abstract method, it's  hook to used in subclass
+        """
+
+    def before_refresh_materialized_view(self, cr, uid, context=None):
+        """Method called before refresh materialized view,
+           this was made to do things like drop index before in the same
+           transaction.
+
+           Nothing done in abstract method, it's  hook to used in subclass
+        """
+
+    def after_refresh_materialized_view(self, cr, uid, context=None):
+        """Method called after refresh materialized view,
+           this was made to do things like add index after refresh data
+
+           Nothing done in abstract method, it's  hook to used in subclass
+        """
+
+    def write(self, cr, uid, ids, context=None):
+        raise osv.except_osv(u"Write on materialized view is forbidden",
+                             u"Write on materialized view is forbidden,"
+                             u"because data would be lost at the next refresh"
+                             )
+
+    def create(self, cr, uid, ids, context=None):
+        raise osv.except_osv(u"Create data on materialized view is forbidden",
+                             u"Create data on materialized view is forbidden,"
+                             u"because data would be lost at the next refresh"
+                             )
+
+    def unlink(self, cr, uid, ids, context=None):
+        raise osv.except_osv(u"Remove data on materialized view is forbidden",
+                             u"Remove data on materialized view is forbidden,"
+                             u"because data would be lost at the next refresh"
+                             )
+
+
+class PGMaterializedViewManager(object):
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def create_mat_view(self, cr, sql, view_name, mat_view_name):
+        """Abstract Method to overwrite in subclass to create sql view
+           and materialized sql view from sql query.
+        """
+
+    @abstractmethod
+    def refresh_mat_view(self, cr, view_name, mat_view_name):
+        """Abstract Method  to overwrite in subclass to refresh
+           materialized sql view
+        """
+
+    @abstractmethod
+    def drop_mat_view(self, cr, view_name, mat_view_name):
+        """Abstract Method to overwrite in subclass to drop materialized view
```

Comment:
I guess we can provide a base implementation here that can be call with super in inherited class